larray/core/metadata.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf8 -*-
+from __future__ import absolute_import, division, print_function
+
+from collections import OrderedDict
+from larray.util.misc import PY2
+
+
+__all__ = ['Metadata']
+
+
+class BaseMetadata(OrderedDict):
+    """
+    Base class for AttributeDict subclass.
+
+    Examples
+    --------
+    >>> import os.path
+    >>> from datetime import datetime
+    >>> from larray.core.array import ndtest
+    >>> tmpdir = getfixture('tmpdir')
+    >>> attrs = Metadata([('string', 'a string'), ('number', 5), ('float', 5.6), ('time', datetime(2017, 2, 10))])
+
+    Save/Load using HDF format
+
+    >>> from larray.util.misc import LHDFStore
+    >>> fname = os.path.join(tmpdir.strpath, 'metadata.h5')
```

Comment:
unfortunately, you cannot import `getfixture` from pytest

Reply:
doh! But AFAICT, you could use the join method to avoid importing os.path:
```python
with LHDFStore(tmpdir.join('metadata.h5').strpath) as store:`
   ...
```

Reply:
OK. But keep in mind that I never plan to add the Metadata class to the API. The examples in metadata.py are only for developers.

Reply:
Make senses... But one should never say never 😄.